python/mxnet/gluon/rnn/rnn_layer.py

```diff
@@ -209,12 +220,16 @@ def _forward_kernel(self, F, inputs, states, **kwargs):
         """ forward using CUDNN or CPU kenrel"""
         if self._layout == 'NTC':
             inputs = F.swapaxes(inputs, dim1=0, dim2=1)
-        params = (kwargs['{}{}_{}_{}'.format(d, l, g, t)].reshape(-1)
-                  for t in ['weight', 'bias']
-                  for l in range(self._num_layers)
-                  for d in ['l', 'r'][:self._dir]
-                  for g in ['i2h', 'h2h'])
-        params = F._internal._rnn_param_concat(*params, dim=0)
+
+        if F is ndarray and self.fused_params is not None:
+            params = self.fused_params
```

Comment:
The hybrid_forward isn't always invoked with the same context, so even if the parameter doesn't change, the context may be wrong.